src/jarabe/journal/journaltoolbox.py

```diff
@@ -727,9 +729,20 @@ def __init__(self):
             menu_box.append_item(button)
 
     def __sort_type_changed_cb(self, widget, property_, icon_name):
+        if(self._property == property_):
+              if(self.order_type=='ascending'):
+                   self.order_type='descending'
+              else:
+                   self.order_type='ascending'
+        else:
+             self.order_type='ascending'
```

Comment:
Can you just directly set the _order property rather than adding the order_type property.